### DISCUSSION TITLE
How to write a right condition expression for an If activity?

### DISCUSSION THREAD

**Discussion by m-asari-sh** [elsa-workflows/elsa-core#3910] [Q&A]
Hi,

Thank you for your great production,

I have created the following test workflow. In the “SwmsCheckVersionStatusServiceActivity” step, I assign a value to a variable and then I use “return Done(result);” to proceed. I expect the workflow to branch to the “Test-True” step if the variable is true and to the “Test-False” step if the variable is false. However, it seems that the condition is not evaluated correctly.

`
                ...

                Activities = new[]
                {
                   ...,
                     new ActivityDefinition
                    {
                        ActivityId = "If",
                        Type = typeof(If).Name,
                        Properties = new List<ActivityDefinitionProperty>
                        {
                           ActivityDefinitionProperty.Liquid("condition", "@input.needToApprove == true")
                        },
                    },
                    new ActivityDefinition
                    {
                        ActivityId = "Test-True",
                        Type = typeof(WriteLine).Name,
                        Properties = new List<ActivityDefinitionProperty>
                        {
                           ActivityDefinitionProperty.Liquid(nameof(WriteLine.Text), "Test-True")
                        }
                    },
                    new ActivityDefinition
                    {
                        ActivityId = "Test-False",
                        Type = typeof(WriteLine).Name,
                        Properties = new List<ActivityDefinitionProperty>
                        {
                           ActivityDefinitionProperty.Liquid(nameof(WriteLine.Text), "Test-False")
                        }
                    },
                },
                Connections = new[]
                {
                    new ConnectionDefinition
                    {
                        SourceActivityId = "SetWorkflowContext",
                        Outcome = OutcomeNames.Done,
                        TargetActivityId = "Test01",
                    },
                    new ConnectionDefinition
                    {
                        SourceActivityId = "Test01",
                        Outcome = OutcomeNames.Done,
                        TargetActivityId = "SwmsCheckVersionStatusServiceActivity",
                    }
                    ,
                    new ConnectionDefinition
                    {
                        SourceActivityId = "SwmsCheckVersionStatusServiceActivity",
                        TargetActivityId = "If",
                        Outcome = OutcomeNames.Done
                    },
                      new ConnectionDefinition
                    {
                        SourceActivityId = "If",
                        TargetActivityId = "Test-True",
                        Outcome = OutcomeNames.True
                    },
                    new ConnectionDefinition
                    {
                        SourceActivityId = "If",
                        TargetActivityId = "Test-False",
                        Outcome = OutcomeNames.False
                    }
                }
            };`
    
 How should I write the condition of the If Step?
 !!!    
 ActivityDefinitionProperty.Liquid("condition", "@input.needToApprove == true")
 !!!

Thank You!

**Comment by m-asari-sh** [comment]
Is there anybody to help?
@sfmskywalker

**Reply by sfmskywalker** [reply]
As a first step to troubleshoot, what happens if you replace the line with this:

```csharp
ActivityDefinitionProperty.Liquid("Condition", "True")
```

Does it work then, or still not?

**Reply by m-asari-sh** [reply]
Thank you for your help. 
Your solution is correct, but I need to verify the result from the previous step("**SwmsCheckVersionStatusServiceActivity**"). 
How can I do this?

**SwmsCheckVersionStatusServiceActivity** is a **UserTask** which returns "Outcome([myBoolVariable])". I want to make a decision in **If** activity by this step result.

If you have another solution, let me know.

Thank you. 

@sfmskywalker

**Reply by sfmskywalker** [reply]
If `SwmsCheckVersionStatusServiceActivity` produces an **outcome** that is either True or False, then you wouldn need to use an If activity - you could connect the two outcomes from SwmsCheckVersionStatusServiceActivity directly to the next activity.

On the other hand, if SwmsCheckVersionStatusServiceActivity produces an **output**, and you want to evaluate that output in an If activity, then the easiest way to do that is by giving the SwmsCheckVersionStatusServiceActivity activity a name so that you can reference it.

For example, if the SwmsCheckVersionStatusServiceActivity has a Name of "SwmsCheckVersionStatusServiceActivity1", then your If activity definition might look like this:

```csharp
new ActivityDefinition
                {
                    ActivityId = "If",
                    Type = typeof(If).Name,
                    Properties = new List<ActivityDefinitionProperty>
                    {
                       ActivityDefinitionProperty.Liquid("condition", "{% if Activities.SwmsCheckVersionStatusServiceActivity1.Output %} True {% else %} False {% endif %}")
                    },
                },
```

But you might want to consider using a JS expression instead, which is easier:

```csharp
new ActivityDefinition
                {
                    ActivityId = "If",
                    Type = typeof(If).Name,
                    Properties = new List<ActivityDefinitionProperty>
                    {
                       ActivityDefinitionProperty.JavaScript("condition", "activities.SwmsCheckVersionStatusServiceActivity1.Output()")
                    },
                },
```

Assuming your activity's `Output` property is a boolean, that JS expression should do the trick.
